### PR TITLE
Initial cleanup before code review

### PR DESCRIPTION
**Pull request by isislovecruft**
There's still a few TODOs but here's an initial set of patches to help ensure production safety.

An incomplete list of things that can/should still be done:

- [ ] Ensure more secret-data types are zeroed out when falling out of scope.
- [ ] Ensure that checks for (rerandomised) public SpendAuth keys are not of small order.
- [ ] Determine if the check for public SpendAuth keys of small order is sufficient, or if we need to ensure that rerandomisers produce group elements of the prime order subgroup. (It's a WIP, but cf. my `feature/frost-key-safety` branch for a start at patching `VerificationKey::from(s: &Scalar)` which is used internally to create and rerandomise SpendAuth signing keys.)
- [ ] Documentation and testing can likely be improved, if external/3rd party adoption is a project goal.

### CODE REVIEW

File: src/frost.rs [hunk 1]
```diff
@@ -16,20 +27,43 @@ use std::convert::TryFrom;
...
-#[derive(Copy, Clone)]
 /// A secret scalar value representing a single signer's secret key.
+#[derive(Clone, Default)]
 pub struct Secret(Scalar);
 
-// TODO: impl From<Scalar> for Secret;
-// TODO: impl From<jubjub::ExtendedPoint> for Public;
+impl Zeroize for Secret {
+    fn zeroize(&mut self) {
+        self.0 = Scalar::zero();
```
Comment (dconnolly) [inline src/frost.rs:41]: Yay!

File: src/frost.rs [hunk 2]
```diff
@@ -16,20 +27,43 @@ use std::convert::TryFrom;
...
+    }
+}
+
+impl Drop for Secret {
+    fn drop(&mut self) {
+        self.zeroize();
```
Comment (dconnolly) [inline src/frost.rs:47]: Yay!

File: src/frost.rs [hunk 3]
```diff
@@ -16,20 +27,43 @@ use std::convert::TryFrom;
...
+    }
+}
+
+impl From<Scalar> for Secret {
```
Comment (dconnolly) [inline src/frost.rs:51]: 🙏 thank you

File: src/frost.rs [hunk 4]
```diff
@@ -149,13 +183,8 @@ pub fn keygen_with_dealer<R: RngCore + CryptoRng>(
...
 
     let secret = Secret(Scalar::from_bytes_wide(&bytes));
-    let group_pk_bytes = jubjub::AffinePoint::from(SpendAuth::basepoint() * secret.0).to_bytes();
-    let group_public = match VerificationKey::try_from(group_pk_bytes) {
-        Ok(x) => x,
-        Err(_) => return Err("Cannot convert public key"),
-    };
-
-    let shares = generate_shares(secret, num_signers, threshold, rng)?;
+    let group_public = VerificationKey::from(&secret.0);
```
Comment (dconnolly) [inline src/frost.rs:186]: Definitely better 😁

**Comment by dconnolly** [comment]
FYI PRs into !main do not automatically have CI run on them currently

**Comment by chelseakomlo** [comment]
This is great. Merging and then we can come back to the remainder of the list post-audit. Thanks @isislovecruft

**Pull request merged**